src/main/java/stream/flarebot/flarebot/Events.java

```diff
@@ -112,15 +114,15 @@ public void onGuildMemberJoin(GuildMemberJoinEvent event) {
                     }
                 }
                 if (welcome.isDmEnabled()) {
-                    String dmMsg = welcome.getRandomDmMessage()
+                    if(event.getMember().getUser().isBot()) return; // We can't DM other bots.
+                    MessageUtils.sendPM(event.getMember().getUser(), welcome.getRandomDmMessage()
                             .replace("%user%", event.getMember().getUser().getName())
                             .replace("%guild%", event.getGuild().getName())
-                            .replace("%mention%", event.getMember().getUser().getAsMention());
-                    MessageUtils.sendPM(event.getMember().getUser(), dmMsg);
+                            .replace("%mention%", event.getMember().getUser().getAsMention()));
                 }
             } else welcome.setGuildEnabled(false);
         }
-        GuildWrapper wrapper = FlareBotManager.getInstance().getGuild(event.getGuild().getId());
+        if (event.getMember().getUser().isBot()) return;
```

Comment:
What is this check even for?

Reply:
We can't give roles to bots so we may as well return before executing that part of the code